lib/posix/timer.c

```diff
@@ -75,6 +107,33 @@ int timer_create(clockid_t clockid, struct sigevent *evp, timer_t *timerid)
 
 	if (evp->sigev_notify == SIGEV_NONE) {
 		k_timer_init(&timer->ztimer, NULL, NULL);
+	} else if (evp->sigev_notify == SIGEV_THREAD) {
+		int ret;
```

Comment:
There is no need to manually allocate a thread stack in this function.

Thread stack allocation is automatic with `pthread_create(&tid, NULL, func, arg)`. I.e. with a `NULL` `pthread_attr_t`.

Reply:
The spec says that users may pass in a `pthread_attr_t *` may be passed in via `sigev_notify_attributes` and if it is non-`NULL` then we should use the stack provided from those attributes.